OtripleS.Web.Api/Models/Classrooms/Exceptions/ClassroomDependencyException.cs

```diff
@@ -4,12 +4,13 @@
 // ---------------------------------------------------------------
 
 using System;
+using Xeptions;
 
 namespace OtripleS.Web.Api.Models.Classrooms.Exceptions
 {
-    public class ClassroomDependencyException : Exception
+    public class ClassroomDependencyException : Xeption
     {
         public ClassroomDependencyException(Exception innerException)
```

Comment:
This input parameter should be of type `Xeption` too.

Reply:
@arafat-pro friendly reminder of this comment.

Reply:
@hassanhabib I need an example to follow and replicate as I am far beyond understanding the total system. Still, I am trying to find out the solution!

Reply:
@arafat-pro all you have to do here is replace `Exception` in the constructor input parameter with `Xeption`.

Reply:
> @hassanhabib 

I've given a try to proceed with replacing `Exception in the constructor input parameter with Xeption.` But cannot find any resolution as more & more errors in the build show up. I've already deployed around 5 hours to deal with this without a solution. May I request you to have a look if possible at [this branch](https://github.com/hassanhabib/OtripleS/commit/e0b4789e6a660ec49f878af3aa3ebb01080cd34c) and suggest to me my course of action it will be very helpful for me!

Reply:
@arafat-pro that means that there are still exceptions out there that don't conform to or implement `Xeption` as a class. We are going to need to find those and have them inherit from `Xeption` unless they are external.

Reply:
> 
I am working on this & hope to resolve it soon now only dependent tests are failing.